scipy/spatial/transform/_rotation.pyx

```diff
@@ -1835,6 +2101,137 @@ cdef class Rotation:
         """
         return self._compute_euler(seq, degrees, 'from_quat')
 
+    @cython.embedsignature(True)
+    def as_davenport(self, axes, extrinsic, degrees=False):
+        """Represent as Davenport angles.
+
+        Any orientation can be expressed as a composition of 3 elementary
+        rotations.
+
+        For Euler angles, we have the following conditions:
+
+            - ``dot(axis1, axis2) = 0``
+            - ``dot(axis2, axis3) = 0``
+            - ``dot(axis1, axis3) = -1``, ``0`` or ``1``
+
+        For Davenport angles, the third relation is relaxed [1]_:
+            - ``dot(axis1, axis3)`` can be any value between -1 and 1.
+
+        A slightly modified version of the algorithm from [2]_ has been used to
+        calculate Davenport angles for the rotation about a given sequence of
+        axes.
+
+        Davenport angles, just like Euler angles, suffer from the problem of
+        gimbal lock [3]_, where the representation loses a degree of freedom
+        and it is not possible to determine the first and third angles
+        uniquely. In this case, a warning is raised, and the third angle is set
+        to zero. Note however that the returned angles still represent the
+        correct rotation.
+
+        Parameters
+        ----------
+        axes : list of (3, 1) arrays
+            Specifies sequence of axes for rotations. Must be a list of
+            3 arrays of shape (3, 1).
+        extrinsic : boolean
+            If True, sequence will be extrinsic. If False, sequence will be
+            treated as intrinsic.
+        degrees : boolean, optional
+            Returned angles are in degrees if this flag is True, else they are
+            in radians. Default is False.
+
+        Returns
+        -------
+        angles : ndarray, shape (3,) or (N, 3)
+            Shape depends on shape of inputs used to initialize object.
+            The returned angles are in the range:
+
+            - First angle belongs to [-180, 180] degrees (both inclusive)
+            - Third angle belongs to [-180, 180] degrees (both inclusive)
+            - Second angle belongs to a set of size 180 degrees, given by:
+            ``[-abs(lambda), 180 - abs(lambda)]``, where ``lambda`` is equal to:
+            ``atan2(dot(axis3, cross(axis1, axis2)), dot(axis3, axis1))``
```

Comment:
Does it correspond to [-90, 90] for Euler angles?

Reply:
It does, because of this operation at the end:
```python
_angles[1] -= lamb
```
For symmetric Euler axes, it's in the range `[0, 180]` (or `[-180, 0]` for what I called non-canonical). In this case, `lamb = 0`, since the angle between the first and third axes is ´0´.

For asymmetric, we have `lamb = 90` degrees, and the range becomes `[90, 90]`.